src/Middleware/Router.php

```diff
@@ -10,25 +10,30 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
-use Yiisoft\Http\Method;
-use Yiisoft\Http\Status;
 use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 use Yiisoft\Middleware\Dispatcher\MiddlewareFactory;
 use Yiisoft\Router\CurrentRoute;
+use Yiisoft\Router\MethodFailureHandler\StandardMethodFailureHandler;
+use Yiisoft\Router\MethodFailureHandler\MethodFailureHandlerInterface;
 use Yiisoft\Router\UrlMatcherInterface;
 
 final class Router implements MiddlewareInterface
 {
     private readonly MiddlewareDispatcher $dispatcher;
+    private readonly MethodFailureHandlerInterface|null $methodFailureHandler;
 
     public function __construct(
         private readonly UrlMatcherInterface $matcher,
-        private readonly ResponseFactoryInterface $responseFactory,
+        ResponseFactoryInterface $responseFactory,
         MiddlewareFactory $middlewareFactory,
         private readonly CurrentRoute $currentRoute,
-        ?EventDispatcherInterface $eventDispatcher = null
+        ?EventDispatcherInterface $eventDispatcher = null,
+        MethodFailureHandlerInterface|false|null $methodFailureHandler = null,
     ) {
         $this->dispatcher = new MiddlewareDispatcher($middlewareFactory, $eventDispatcher);
+        $this->methodFailureHandler = $methodFailureHandler === false
+            ? null
+            : $methodFailureHandler ?? new StandardMethodFailureHandler($responseFactory);
```

Comment:
1. What's the goal of not having any handler?
2. Ternary condition here is hardly readable. Better to split it.